File: fotoapparat/src/main/java/io/fotoapparat/util/CompareFpsRangeByBounds.java
package io.fotoapparat.util;

import java.util.Comparator;

import io.fotoapparat.parameter.range.Range;

/**
 * Comparator based on bounds check. Lower bound has higher priority.
 */
public class CompareFpsRangeByBounds implements Comparator<Range<Integer>> {

    @Override
    public int compare(Range<Integer> fpsRange1, Range<Integer> fpsRange2) {
        int lowerBoundCompare = fpsRange1.lowest().compareTo(fpsRange2.lowest());
        if (lowerBoundCompare == 0) {
            int upperBoundCompare = fpsRange1.highest().compareTo(fpsRange2.highest());
            return upperBoundCompare;
        }
        return lowerBoundCompare;
    }
}

File: fotoapparat/src/main/java/io/fotoapparat/util/CompareFpsRangeByBounds.kt
package io.fotoapparat.util

import io.fotoapparat.parameter.FpsRange
import java.util.*

/**
 * Comparator based on bounds check. Lower bound has higher priority.
 */
internal object CompareFpsRangeByBounds : Comparator<FpsRange> {

    override fun compare(fpsRange1: FpsRange, fpsRange2: FpsRange): Int {
        return fpsRange1.min.compareTo(fpsRange2.min).run {
            when (this) {
                0 -> fpsRange1.max.compareTo(fpsRange2.max)
                else -> this
            }
        }
    }
}